api/handler/v1beta1/job_spec.go

```diff
@@ -163,30 +163,44 @@ func (sv *JobSpecServiceServer) CreateJobSpecification(ctx context.Context, req
 		return nil, mapToGRPCErr(sv.l, err, "unable to get namespace")
 	}
 
-	jobSpec, err := FromJobProto(req.GetSpec(), sv.pluginRepo)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
+	reqJobSpecs := req.GetSpecs()
+	var jobSpecs []models.JobSpec
+	for _, spec := range reqJobSpecs {
+		jobSpec, err := FromJobProto(spec, sv.pluginRepo)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
+		}
+		jobSpecs = append(jobSpecs, jobSpec)
+	}
+
+	jobNames := make([]string, len(jobSpecs))
+	for i, jobSpec := range jobSpecs {
+		jobNames[i] = jobSpec.Name
 	}
 
 	// validate job spec
-	if err = sv.jobSvc.Check(ctx, namespaceSpec, []models.JobSpec{jobSpec}, logWriter); err != nil {
+	if err = sv.jobSvc.Check(ctx, namespaceSpec, jobSpecs, logWriter); err != nil {
```

Comment:
How about call to the JobService only once, so we will put the flow of having to do checking, saving, and deployment scheduling in JobService?
- We don't need to export IdentifyAndPersistJobSources, BulkCreate & ScheduleDeployment
- Since JobService.Create already exist, we need to think what should be the proper name of this new function & the existing Create function